apps/explorer/src/components/ownedobjects/views/ObjectFieldsCard.tsx

```diff
@@ -0,0 +1,249 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Combobox } from '@headlessui/react';
+import { getObjectFields } from '@mysten/sui.js';
+import clsx from 'clsx';
+import { useState } from 'react';
+
+import { ReactComponent as SearchIcon } from '~/assets/SVGIcons/24px/Search.svg';
+import { SyntaxHighlighter } from '~/components/SyntaxHighlighter';
+import { useGetObject } from '~/hooks/useGetObject';
+import { DisclosureBox } from '~/ui/DisclosureBox';
+import { AddressLink, ObjectLink, TransactionLink } from '~/ui/InternalLink';
+import { LoadingSpinner } from '~/ui/LoadingSpinner';
+import { Tab, TabGroup, TabList, TabPanel, TabPanels } from '~/ui/Tabs';
+import { Text } from '~/ui/Text';
+import { ListItem, VerticalList } from '~/ui/VerticalList';
+
+interface ObjectFieldsProps {
+    id: string;
+}
+
+interface FieldItemProps<T> {
+    value: T;
+    type?: string;
+}
+
+function FieldItem<T>({ value, type }: FieldItemProps<T>) {
+    if (typeof value === 'object') {
+        return (
+            <SyntaxHighlighter
+                code={JSON.stringify(value, null, 2)}
+                language="json"
+            />
+        );
+    }
+    if (typeof value === 'string' && type === 'address') {
+        return <AddressLink address={value} />;
+    }
+
+    if (typeof value === 'string' && type === 'objectId') {
+        return (
+            <div className="break-all">
+                <ObjectLink objectId={value} />
+            </div>
+        );
+    }
+
+    if (typeof value === 'string' && type === 'digest') {
+        return <TransactionLink digest={value} />;
+    }
+
+    return (
+        <Text variant="body/medium" color="steel-darker">
+            {value?.toString()}
+        </Text>
+    );
+}
+
+export function ObjectFieldsCard({ id }: ObjectFieldsProps) {
+    const { data, isLoading, isError } = useGetObject(id);
+    const [query, setQuery] = useState('');
+    const [activeFieldName, setActiveFieldName] = useState('');
+
+    if (isLoading) {
+        return (
+            <div className="flex w-full justify-center">
+                <LoadingSpinner text="Loading data" />
+            </div>
+        );
+    }
+    if (isError) {
+        return null;
+    }
+
+    const fieldsData = getObjectFields(data!);
+    const fieldsNames = Object.entries(fieldsData || {});
+    const filteredFieldNames =
+        query === ''
+            ? fieldsNames
+            : fieldsNames
+                  .filter(([name]) =>
+                      name.toLowerCase().includes(query.toLowerCase())
+                  )
+                  .map((name) => name);
+
+    // TODO scroll to active field
+    const onChangeField = (newKeyName: string) => {
+        setActiveFieldName(newKeyName);
+    };
+
+    return fieldsNames?.length && fieldsData ? (
```

Comment:
I'd probably change this to use an `if` with an early return just to keep the ternary out of the return.